MicrosoftStoreServicesSample/ClawbackManager/ClawbackManager.cs

```diff
@@ -116,6 +116,41 @@ private async Task RemoveClawbackQueueItemFromTrackingIdAsync(string trackingId,
             }
         }
 
+        /// <summary>
+        /// Update the Queue item's PurchaseId to a refreshed one
+        /// </summary>
+        /// <param name="TrackingId"></param>
+        /// <param name="NewKeyPurchaseId"></param>
+        /// <param name="cV"></param>
+        /// <returns></returns>
+        private async Task UpdateClawbackQueueItemPurchaseId(String TrackingId, String NewKeyPurchaseId, CorrelationVector cV)
+        {
+            try
+            {
+                using (var dbContext = ServerDBController.CreateDbContext(_config, cV, _logger))
+                {
+                    //  We should have a unique LineItemId
+                    var item = dbContext.ClawbackQueue.Find(TrackingId);
+                    
+                    if (item != null)
+                    {
+                        item.UserPurchaseId = NewKeyPurchaseId;
+                        await dbContext.SaveChangesAsync();

```

Comment:
I don't think this SaveChangesAsync is required since it's called regardless below.

Reply:
removed the save towards the bottom so it only happens if it needs to.